client/src/components/shop/ProductCard.js

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import Typography from "@material-ui/core/Typography";
+import Card from "@material-ui/core/Card";
+import CardContent from "@material-ui/core/CardContent";
+import CardMedia from "@material-ui/core/CardMedia";
+import SettingsIcon from "@material-ui/icons/Settings";
+import FavoriteBorderOutlinedIcon from "@material-ui/icons/FavoriteBorderOutlined";
+import FavoriteIcon from "@material-ui/icons/Favorite";
+import { makeStyles } from "@material-ui/core/styles";
+import Grid from "@material-ui/core/Grid";
+
+const useStyles = makeStyles(() => ({
+  card: {
+    height: "100%",
+    display: "flex",
+    flexDirection: "column"
+  },
+  cardMedia: {
+    paddingTop: "56.25%" // 16:9
+  },
+  cardContent: {
+    flexGrow: 1,
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    backgroundColor: "black",
+    color: "white"
+  },
+  cardSettingBtn: {
+    position: "relative",
+    left: "100%",
+    transform: "translate(-100%,0)",
+    "&:hover": {
+      color: "grey"
+    }
+  },
+  cardLikedBtn: {
+    position: "relative",
+    transform: "translate(-100%,0)",
+    "&:hover": {
+      color: "grey"
+    }
+  },
+  cardHeader: {
+    height: "0px",
+    width: "100%"
+  }
+}));
+
+export default function ProductCard(props) {
+  const classes = useStyles();
+  return (
+    <Grid item xs={12} sm={6} md={4}>
+      <Card className={classes.card}>
+        <div className={classes.cardHeader}>
+          <SettingsIcon className={classes.cardSettingBtn} />
+          <FavoriteIcon
+            className={classes.cardLikedBtn}
+            style={
+              props.product.isLiked === false
+                ? { display: "none" }
+                : { display: "initial" }
+            }
+            onClick={() =>
+              props.dispatch({ type: "PRODUCT_LIKED", likeIndex: props.index })
```

Comment:
In the future this will make API calls?

Reply:
I think we will have to store what a user has liked in their user model. We would likely have a page where they can view their liked content. That is the point of likes or wishlist i think.